### ISSUE TITLE
Add endpoints to list and accept membership requests

### ISSUE THREAD

**Issue by donni106** [ikuseiGmbH/smart-village-app-humhub-ext#15]
As a follow up for #3 i wonder if there are already endpoints for listing all membership requests for a space and an endpoint for accepting them? An owner of a space should be able to list and accept membership requests.

**Comment by gaurav-handysolver** [comment]
Hi @donni106 
We checked in space module and there are two functions exist, one for pending approval for a space and another one is to accept all the pending approvals. But there is no endpoint seems to exist for them.
Thanks!

**Comment by gaurav-handysolver** [comment]
Hi @donni106 
Do you want us to create both these endpoints?
First one is listing of all requested membersihp and second one is to accpet those pending approval.
Thanks!

**Comment by donni106** [comment]
Yes please. The approval should work one by one for each membership request. Do you need any further information?

**Comment by gaurav-handysolver** [comment]
Hi @donni106 
Thanks for the reply. we will work on this.

 We suggesting to you some of the endpoints URLs.
    **for list of all requested membership**
           /space/{space-id}/list/request/membership
          /space/{space-id}/list/request
         /space/{space-id}/list/request-membership

    **for approved each membership request**
             /space/{space-id}/accept/{userId}
          /space/{space-id}/accept-membership/{userId}

You can feel free to share your endpoint URLs to us.
Thanks!

**Comment by donni106** [comment]
Thanks for the question, as there are already "membership scopes" with `/space/{spaceId}/membership` and `/space/{spaceId}/membership/{userId}/...` we should stay in that structure. With that in mind i suggest:

**list of all membership requests**
`/space/{spaceId}/membership/requests`

but with checking the result of `/space/{spaceId}/membership`, every entry has a `status: 1 | 2 | 3` (1=Invited, 2=Applicant, 3=Member). `status: 2` are the ones with membership requests. So, we will not need a new endpoint, right? All iformation for a list of requests is already available.

**accepting a membership request**
after that i have tried to use another existing endpoint `POST /space/{spaceId}/membership/{userId}` to add a member with `status: 2` to the space. and yes, it worked.

so, we do not need new endpoints or am i wrong?

**Comment by gaurav-handysolver** [comment]
Hi @donni106 
We investigated the above endpoints and we found.

1. list of all membership request (/space/{spaceId}/membership)
   When we hit the above endpoint url , it is fetching both member (applicant as well as member) user.
![image](https://user-images.githubusercontent.com/103625795/167358864-18a5c949-2264-4ead-b5a7-a14949b99476.png)

2. accepting a membership request (/space/{spaceId}/membersip/{userId})
   And when we hit the above endpint url to add a member in the space, it is adding member with status : 3 (which is Member not Applicant).
![image](https://user-images.githubusercontent.com/103625795/167359582-e256dd0a-55fb-4caa-a733-c7da4f8afa7f.png)

The last one
![image](https://user-images.githubusercontent.com/103625795/167359657-dfa8541f-7e32-4d7e-b4dd-0f64679dd73d.png)

Thanks!
